truss/templates/trtllm/model/model.py

```diff
@@ -30,70 +27,48 @@ def __init__(self, data_dir, config, secrets):
         self.uses_openai_api = None
 
     def load(self):
-        execute_command(["ldconfig"])
-        trtllm_config = TrussTRTLLMConfiguration(**self._config.get("trt_llm", {}))
+        trtllm_config = self._config.get("trt_llm", {})
         self.uses_openai_api = "openai-compatible" in self._config.get(
             "model_metadata", {}
         ).get("tags", [])
         hf_access_token = None
         if "hf_access_token" in self._secrets._base_secrets.keys():
             hf_access_token = self._secrets["hf_access_token"]
 
-        # The underlying engine build call is idempotent, so we can call it in the load
-        # even if the engine is already built. The engine build call checks to see if
-        # there are .engine files in the destination directory and skips the build if
-        # they are present.
-        if trtllm_config.requires_build:
-            build_engine_utils.build_engine_from_config_args(
-                truss_trtllm_configuration=trtllm_config,
-                checkpoint_dir_path=None,
-                dst=self._data_dir,
-            )
+        engine_repository_path = self._data_dir
```

Comment:
Can we try to cast this to the appropriate underlying type and use a Pydantic type to parse out the values we care about? The underlying base image doesn't contain the types anymore, is there another way to pass that into the truss?

Reply:
yeah will try and make truss a dep of this example and leverage the type there